### PR TITLE
Cleanup unused imports and variables in backends

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Remove unused imports and local variables.

I split some import lines into two: one with the used and one with the unused. In this way, tools will warn if any previously used import becomes unused.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/backends/backend_agg.py [hunk 1]
```diff
@@ -29,11 +29,10 @@
 from math import radians, cos, sin
 
 import numpy as np
-from PIL import Image
+from PIL import Image  # noqa: F401 # pylint: disable=W0611
```
Comment (anntzer) [inline lib/matplotlib/backends/backend_agg.py:32]: just remove the import?

File: lib/matplotlib/backends/backend_qt.py [hunk 2]
```diff
@@ -405,8 +405,7 @@ def start_event_loop(self, timeout=0):
             raise RuntimeError("Event loop already running")
         self._event_loop = event_loop = QtCore.QEventLoop()
         if timeout > 0:
-            timer = QtCore.QTimer.singleShot(int(timeout * 1000),
-                                             event_loop.quit)
```
Comment (anntzer) [inline lib/matplotlib/backends/backend_qt.py:409]: Did you check that it's OK to get the timer get gc'ed immediately?  e.g. what happens if you insert a gc.collect() right here (gc'ing the timer); does a e.g. 2s timer get run?
Perhaps it is OK, but it's never obvious with GUI toolkits...

Comment (oscargus) [reply lib/matplotlib/backends/backend_qt.py:409]: Not really. I'll replace it with `_`.

File: lib/matplotlib/backends/backend_agg.py (hunk 1 above)
Comment (oscargus) [reply lib/matplotlib/backends/backend_agg.py:32]: Yes. I thought removing  it caused some local problems, but I was probably wrong.

File: lib/matplotlib/backends/backend_tkagg.py [hunk 3]
```diff
@@ -1,7 +1,8 @@
 from . import _backend_tk
 from .backend_agg import FigureCanvasAgg
-from ._backend_tk import (
-    _BackendTk, FigureCanvasTk, FigureManagerTk, NavigationToolbar2Tk)
+from ._backend_tk import _BackendTk, FigureCanvasTk
+from ._backend_tk import (  # noqa: F401 # pylint: disable=W0611
+                          FigureManagerTk, NavigationToolbar2Tk)
```
Comment (QuLogic) [inline lib/matplotlib/backends/backend_tkagg.py:5]:
```suggestion
    FigureManagerTk, NavigationToolbar2Tk)
```

**Review by QuLogic — APPROVED** [review]

**Review by timhoffm — APPROVED** [review]

**Pull request merged**